src/components/global/Playground/index.tsx

```diff
@@ -401,73 +368,46 @@ export default function Playground({
   });
 
   /**
-   * By default, we do not render the iframe content
-   * as it could cause delays on page load. Instead
-   * we wait for even 1 pixel of the playground to
-   * scroll into view (intersect with the viewport)
-   * before loading the iframes.
+   * By default, we do not render the iframe content as it could
+   * cause delays on page load. We also do not immediately switch
+   * the framework/mode when it gets changed through another
+   * playground on the page, as switching them for every playground
+   * at once can cause memory-related crashes on some devices.
+   * 
+   * Instead, we wait for even 1 pixel of the playground to scroll
+   * into view (intersect with the viewport) before loading the
+   * iframes or auto-switching the framework/mode.
    */
   useEffect(() => {
     const io = new IntersectionObserver(
       (entries: IntersectionObserverEntry[]) => {
         const ev = entries[0];
-        if (!ev.isIntersecting || renderIframes) return;
+        if (!ev.isIntersecting) return;
 
-        setRenderIframes(true);
+        /**
+         * Load the stored mode and/or usage target, if present
+         * from previously being toggled.
+         */
+        if (isBrowser) {
+          const storedMode = localStorage.getItem(MODE_STORAGE_KEY);
+          if (storedMode) setIonicMode(storedMode);
+          const storedUsageTarget = localStorage.getItem(USAGE_TARGET_STORAGE_KEY);
+          if (storedUsageTarget) setUsageTarget(storedUsageTarget);
+        }
 
         /**
-         * Once the playground is loaded, it is never "unloaded"
-         * so we can safely disconnect the observer.
+         * If the iframes weren't already loaded, load them now.
          */
-        io.disconnect();
+        if (!renderIframes) {
+          setRenderIframes(true);
+        }
       },
       { threshold: 0 }
     );
 
     io.observe(hostRef.current!);
   });
 
-  /**
-   * Sometimes, the app isn't fully hydrated on the first render,
-   * causing isBrowser to be set to false even if running the app
-   * in a browser (vs. SSR). isBrowser is then updated on the next
-   * render cycle.
-   *
-   * This useEffect contains code that can only run in the browser,
-   * and also needs to run on that first go-around. Note that
-   * isBrowser will never be set from true back to false, so the
-   * code within the if(isBrowser) check will only run once.
-   */
-  useEffect(() => {
```

Comment:
Loading the mode/framework when the playground comes into view also covers playgrounds that were in view from the start (such as the [Button page](https://ionicframework.com/docs/api/button)), so this entire `useEffect` is no longer needed.